src/v/storage/tests/storage_e2e_test.cc

```diff
@@ -1734,7 +1734,8 @@ FIXTURE_TEST(disposing_in_use_reader, storage_test_fixture) {
         ss::sleep(200ms).get();
     }
     // yield again to allow truncate fiber to finish
-    ss::sleep(200ms).get();
+    ss::sleep(100ms).get();
+    ss::sleep(100ms).get();
```

Comment:
Patch seems fine to me but just a question, any reason we don't use something like cooperative_spin_wait_with_timeout(large_timeout, () -> truncate_f.available()) to be more non-deterministic here?

Reply:
The primitive that I would really want for tests like these is a "drain" on the reactor: the expectation is that all ready futures should complete, rather than that any particular length of time should elapse.

I looked into this a while ago and it looks like there isn't a neat way of doing it without modifying seastar (but maybe we should do that).

Reply:
I see, thanks for explaining.

> rather than that any particular length of time should elapse.

Doesn't cooperative_spin_wait_with_timeout() do exactly what you are trying to do, just that it wraps it in a loop with 10ms yield after every check. The reason I thought yielding for a couple of times is non-deterministic is because we are at the mercy of the ss scheduler and things might go wrong in some crazy test env.